pkg/controller/nodes/task/k8s/plugin_manager.go

```diff
@@ -102,12 +102,15 @@ type PluginManager struct {
 }
 
 func (e *PluginManager) AddObjectMetadata(taskCtx pluginsCore.TaskExecutionMetadata, o client.Object, cfg *config.K8sPluginConfig) {
-	o.SetNamespace(taskCtx.GetNamespace())
+	if len(o.GetNamespace()) == 0 {
```

Comment:
cc @hamersaw Is there any concern to do this?

Reply:
Maybe make it opt-in. like
```
if !K8sPluginConfig.namespaceOverride || len(o.GetNamespace()) == 0 {
    o.SetNamespace(taskCtx.GetNamespace())
}
```

Reply:
This seems very hacky to me. I think the next obvious ask is going to be to support this for all plugins. So rather than adding an option to each config object (ie. Ray, Dask, Spark, Pod, etc) is there a way we can add this override to the task execution context so that it's applicable to all k8s plugin types?